### PR TITLE
Improve the getic.sh to retrive restart files for warm start for GEFS

### LINKED ISSUE #911: Extend coupled IC job to handle ensemble staging
**Description**
Allow staging of ensemble initial conditions for GEFS through the `coupled_ic.sh` job for `forecast-only` mode.

**Requirements**
- create  a j-job `JGLOBAL_STAGE_IC` and ex-script `exglobal_stage_ic.sh` to allow staging of initial conditions for GFS and GEFS **forecast-only** experiments.
- `coupled_ic.sh` (rename to `stage_ic.sh`) should be the rocoto wrapper on `JGLOBAL_STAGE_IC`.

The initial conditions should be ready to go -- i.e. all pre-processing has been done prior to staging.

**Acceptance Criteria (Definition of Done)**
Call sequence:
`stage_ic.sh -> JGLOBAL_STAGE_IC -> exglobal_stage_ic.sh`

**Dependencies**
May need an update in the files situated under `BASE_CPLIC` that serves as a repository for ICs.

### PR DESCRIPTION
**Pull request by XianwuXue-NOAA**
**Description**

This PR includes two sets of changes related to merging GEFS into the GFS system:
1) Change the DATA to include jobid to avoid different members using same DATA
2) Add new features to retrieve GFS and ensemble members restart files and stage them in the appropriate directories

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

The script was tested on Hera.

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

Refs: #911

### CODE REVIEW

File: jobs/rocoto/getic.sh [hunk 1]
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
...
+            rm -rf gfs.${yy}${mm}${dd}
+            cd $sPath_out
+            for f in `ls gfs.*`
+            do
+                echo $f
+                fnew=${f/gfs/gefs}
+                mv $f $fnew
+            done
+            cd RESTART
+            module load nco/4.9.3
```
Comment (aerorahul) [inline jobs/rocoto/getic.sh]: The `nco` module is loaded in `module_base.$target`.

File: jobs/rocoto/getic.sh [hunk 2]
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
...
+            rm -rf gdas.${gyy}${gmm}${gdd}
+            cd $sPath_out
+            for f in `ls gdas.*`
+            do
+                echo $f
+                fnew=${f/gdas/gefs}
+                mv $f $fnew
+            done
+            cd RESTART
+            module load nco/4.9.3
```
Comment (aerorahul) [inline jobs/rocoto/getic.sh]: same comment as above.

File: jobs/rocoto/getic.sh [hunk 3]
```diff
@@ -82,11 +82,134 @@ if [[ $gfs_ver = "v16" && $EXP_WARM_START = ".true." && $CASE = $OPS_RES ]]; the
...
+            elif [ $cmem -gt 40 ]; then
+                sgrp=5
+            elif [ $cmem -gt 30 ]; then
+                sgrp=4
+            elif [ $cmem -gt 20 ]; then
+                sgrp=3
+            elif [ $cmem -gt 10 ]; then
+                sgrp=2
+            elif [ $cmem -gt 0 ]; then
+                sgrp=1
```
Comment (aerorahul) [inline jobs/rocoto/getic.sh]: seems like this can be reduced to `cmem/10` and/or `cmem%10` rather than this big `if-else` block.

**Review by aerorahul — CHANGES_REQUESTED** [review]
In principle the changes are fine.
A few minor comments apart from those in the review:
- remove the `echo` where unnecessary and all the `ls` statements.
- module loads need to go out from this script

A major comment:
- a lot of the code is duplicated.  Can we work to reduce that?

File: jobs/rocoto/getic.sh (hunk 1 above)
Comment (XianwuXue-NOAA) [reply jobs/rocoto/getic.sh]: OK, I will delete this line.

File: jobs/rocoto/getic.sh (hunk 2 above)
Comment (XianwuXue-NOAA) [reply jobs/rocoto/getic.sh]: No problem. I will delete this one.

**Comment by XianwuXue-NOAA** [comment]
@aerorahul I am working on reducing the duplicated statements?

**Comment by XianwuXue-NOAA** [comment]
@aerorahul I have modified the script and pushed to this PR

**Pull request closed without merging**